### ISSUE TITLE
tf.contrib.distribute performance issue with asymmetrical multi GPU setup

### ISSUE THREAD

**Issue by peidaqi** [tensorflow/tensorflow#25425]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 16.04, Kernel 4.13.16
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): tensorflow-gpu binary from pip
- TensorFlow version (use command below): 1.12.0
- Python version: 3.6.3
- Bazel version (if compiling from source): N/A
- GCC/Compiler version (if compiling from source): N/A
- CUDA/cuDNN version: CUDA 9.0 with NCCL 2.3.7-1
- GPU model and memory: 1080ti 12G + 1070ti 8G

**Description**
I've been trying to port the Google BERT language model to support multi-GPU transfer learning using the new tf.contrib.distribute API. My hardware setup is a 1070ti + 1080ti, and it's a bit strange because I'm getting similar performance between the dual-GPU and a single 1070ti.

To reproduce this issue I built a much simpler model (see link) with a timer. And here's the results I got (in secs):
3.0106631300004665 1070Ti + 1080Ti
2.8305218839959707 1070Ti
1.6665251980011817 1080Ti

I can imagine the 1080Ti being throttled back with a MirroredStrategy, but at least it should perform like a 1070Ti and with the two cards gives 1.6~1.8X of the performance of a 1070Ti? Maybe this is a load-balancing issue with the NCCL?

**Code to reproduce the issue**
https://drive.google.com/file/d/15nJlVV9PWzKcF1AI24H2XSfzzCAhwQ5f/view?usp=sharing

(Change line 30 to specify different GPUs)

**Other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

**Comment by guptapriya** [comment]
@peidaqi thank you for trying out MirroredStrategy. 
I can't comment on what you're seeing with BERT without more information. But for the small example you provided - my hypothesis is that the graph building time is likely majority of the time that you measured. MirroredStrategy has a known overhead at graph building time as it tries to build 2 copies of your computations, one for each replica, and it does this sequentially. Once this is done, however, the training itself should be faster with 2 comparable GPUs than with one. 

One thing to remember when you're comparing performances is that your dataset will get consumed in half the number of steps in 2 replica case, as with 1. So if you're using steps / s as the measurement of performance when looking at BERT, then you should expect to get similar or slightly lower number of steps as you increase #GPUs (because you're just consuming more data in each step).

Let me know if these hypothesis match up with what you see.

**Comment by mohantym** [comment]
Hi @peidaqi!
It seems you are using older versions(1.x versions) of Tensorflow which is not supported any more. Please upgrade your code base to 2.7 and create a new issue if you need further assistance.Thanks!
